### PR TITLE
Don't change part layout when fixing object metadata

### PR DESCRIPTION
**Pull request by xbrianh**
This fixes a bug in storage operations that causes part layout to change when fixing metadata on multipart objects in s3.

connected to #2073

### CODE REVIEW

File: environment [hunk 1]
```diff
@@ -54,26 +54,26 @@ DSS_S3_BUCKET_TEST=org-hca-dss-test
 DSS_S3_BUCKET_TEST_FIXTURES=org-hca-dss-test-fixtures
 DSS_S3_BUCKET_INTEGRATION=org-humancellatlas-dss-integration
 DSS_S3_BUCKET_STAGING=org-humancellatlas-dss-staging
-DSS_S3_BUCKET_PROD=org-humancellatlas-dss-prod
+DSS_S3_BUCKET_PROD=org-hca-dss-prod
```
Comment (DailyDreaming) [inline environment]: Looks like this file might have been changed accidentally?

File: dss/operations/storage.py [hunk 2]
```diff
@@ -146,12 +158,16 @@ def process_key(self, key):
                     update_aws_content_type(client, self.replica.bucket, blob_key, file_metadata['content-type'])
                 elif Replica.gcp == self.replica:
                     update_gcp_content_type(client, self.replica.bucket, blob_key, file_metadata['content-type'])
+            elif Replica.aws == self.replica:
+                blob_etag = self.handle.get_cloud_checksum(self.replica.bucket, blob_key)
+                if blob_etag != file_metadata['s3-etag']:
+                    update_aws_content_type(client, self.replica.bucket, blob_key, file_metadata['content-type'])
```
Comment (DailyDreaming) [inline dss/operations/storage.py]: Should this `assert` that the new etag matches the metadata etag on the next line?

File: environment (hunk 1 above)
Comment (xbrianh) [reply environment]: will fix

File: dss/operations/storage.py (hunk 2 above)
Comment (xbrianh) [reply dss/operations/storage.py]: will add

**Pull request merged**